meegpowreg/pipelines.py

```diff
@@ -17,126 +17,147 @@
-            vec_naive,
-            expander,
-            StandardScaler(),
-            RidgeCV(alphas=ridge_alphas)),
-        'spoc': make_pipeline(
-            vec_spoc,
-            expander,
-            StandardScaler(),
-            RidgeCV(alphas=ridge_alphas)),
-        'riemann_wass': make_pipeline(
-            vec_riemann_wass,
-            expander,
-            StandardScaler(),
-            RidgeCV(alphas=ridge_alphas)),
-        'dummy': make_pipeline(
-            vec_logdiag,
-            expander,
-            StandardScaler(),
-            DummyRegressor())
+def make_filter_bank_model(names, method='riemann', projection_params=None,
+                           vectorization_params=None,
+                           categorical_interaction=None, scaling=None,
+                           estimator=None):
+    """Generate pipeline for filterbank models.
+
+    Compute filter bank models as used in [1]_. These models take as input
+    sensor-space covariance matrices computed from M/EEG signals in different
+    frequency bands. Then transformations are applied that can improve the
+    applicability of linear regression techniques by removing field spread.
+
+    In terms of implementation, this involves 1) projection
+    (e.g. spatial filters) and 2) vectorization (e.g. taking the log on the
+    diagonal).
+
+    .. note::
+        The resulting model expects as inputs data frames in which different
+        covarances (e.g. for different frequencies) are stored inside columns
+        indexed by ``names''.
+
+        Other columns will be passed through by the underlying column
+        transformers.
+
+        The pipeline also supports fitting categorical interaction effects
+        after projection and vectorization steps are performed.
+
+    .. note::
+        All essential methods from [1]_ are implemented here. In practice,
+        we recommend comparing `riemann', `spoc' and `diag' as a baseline.
+
+    Parameters
+    ----------
+    names : list of str
+        The column names of the data frame corresponding to different
+        covariances.
+    method : str
+        The method used for extracting features from covariances. Defaults
+        to `riemann'.
+    projection_params : dict
+        The parameters for the projection step.
+    vectorization_params : dict
+        The parameters for the vectorization step.
+    categorical_interaction : str
+        The column in the input data frame containing a binary descriptor
+        used to fit 2-way interaction effects.
+    scaling : scikit-learn Transformer object | None
+        Method for re-rescaling the features. Defaults to None. If None,
+        StandardScaler is used.
+    estimator : scikit-learn Estimator object.
+        The estimator object. Defaults to None. If None, RidgeCV is used with
+        generalized cross validation for the regularization parameter alpha.
+        A logarithmic space between -3 and 5 is visited (100 values).
```

Comment:
I am wondering if we should not make 3 functions

make_filter_bank_transformer
make_filter_bank_regressor
make_filter_bank_classifier

where make_filter_bank_transformer returns the first steps of the pipeline without the scaler and the RIdgeCV

Reply:
I was thinking in that direction too. `make_filter_bank_regressor/classifier` would then just be high-level convenience functions that work API-wise like the current `make_filter_bank_model`?